packages/ssz/src/type/basic.ts

```diff
@@ -30,11 +30,18 @@ export abstract class BasicType<V> extends Type<V> {
   }
 
   hashTreeRoot(value: V): Uint8Array {
-    // TODO: Optimize
-    const uint8Array = new Uint8Array(32);
+    // cannot use allocUnsafe() here because hashTreeRootInto() may not fill the whole 32 bytes
```

Comment:
is this comment correct? it looks like hashTreeRootInto is filling all 32 bytes

Reply:
I'd remove that comment @nazarhussain . I remember I tried `allocUnsafe()` at some points and got "Invalid state root" issue. If we want to apply it, need to audit carefully an apply to other places as well